### ISSUE TITLE
After 43 upgrade, all scans crash if debugger not connected

### ISSUE THREAD

**Issue by cvburgess** [expo/expo#15040]
### Summary

We used 42 just fine, but while attempting to upgrade to 43 the development client will only successfully run the app code if the remote debugger is enabled. When scanning QR codes from PRs where the debugger is not an option, the app never loads.

### Managed or bare workflow? If you have made manual changes inside of the `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

iOS

### Package versions

```
"expo-dev-client": "~0.6.3",
"expo-updates": "~0.10.10",
```

### Environment

```
Expo CLI 4.12.11 environment info:
    System:
      OS: macOS 12.0.1
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 16.2.0 - /usr/local/bin/node
      npm: 7.20.5 - /usr/local/bin/npm
      Watchman: 2021.08.23.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.10.2 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.0.1, iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0
    IDEs:
      Android Studio: 4.2 AI-202.7660.26.42.7351085
      Xcode: 13.1/13A1030d - /usr/bin/xcodebuild
    npmPackages:
      @expo/webpack-config: ^0.16.2 => 0.16.6 
      expo: ^43.0.0 => 43.0.1 
      react: 17.0.1 => 17.0.1 
      react-dom: 17.0.1 => 17.0.1 
      react-native: 0.64.2 => 0.64.2 
      react-native-web: 0.17.1 => 0.17.1 
    npmGlobalPackages:
      expo-cli: 4.12.11
    Expo Workflow: managed
    ```

### Reproducible demo or steps to reproduce from a blank project

Using EAS for builds or local start command.

If Debug Remote JS is enabled, we can run the app.
If it is disabled we either don't get past the splash screen or get a blue screen with a debugger-related error.

**Comment by brentvatne** [comment]
would it be possible to post a reproducible example please? something we can clone and run to see this issue

**Comment by cvburgess** [comment]
Woof, I'm pretty slammed - it may take me a bit to create a new project.

Is this not happening to anyone else? Are there config files I can share or something short of a whole build I can provide?

Our 42 app has been working fine for months, we upgraded to 43 following the exact steps in the blog post, and hit these issues.

**Comment by dackom** [comment]
I am having similar issue. After update to expo 43, I have random crashes on iOS mostly on app start (in first few seconds), when running fully compiled app trough test flight. It never happens on cable and when connected to metro so I can't actually catch any log. App just crashes, exits, with no message.

And to be sure that is not javascript related, it happens even when app file is just <View><Text>Blah</Text></View> and nothing else imported at the top. I am suspecting that is somehow related to expo-updates, so I am now trying to build iOS version where EXUpdatesCheckOnLaunch is NEVER .

**Comment by cvburgess** [comment]
I think I may have isolated it to bumping the metro config that the CLI recommended upgrading but wasn't automatically upgraded.

```
"@expo/metro-config": "^0.2.7", // Broken
"@expo/metro-config": "^0.1.84", // Appears to work
```

When I revert the metro config, I can again build locally, but when I scan a QR code in a PR or otherwise hosted on expo's servers, I get this error in Sentry:

```
NSRangeException

Cannot remove an observer <EXSplashScreenService 0x281c13180> for the key path "rootViewController" from <EXDevMenu.DevMenuWindow 0x10251a2c0> because it is not registered as an observer.
```

**Comment by brentvatne** [comment]
@cvburgess - can you try using `expo-splash-screen@0.13.5`?

**Comment by cvburgess** [comment]
@brentvatne its not in our package.json, though it is in our lockfile.

I can set it explicitly to 0.13.5, right now we have 0.13.3 in the lockfile.

I will report back in ~30 mins or so once EAS can cut a new dev client and I can test it.

Here is the full sentry issue: https://sentry.io/share/issue/c05e8566798444b2b74de6cf8a95d5cf/

**Comment by brentvatne** [comment]
also, if you use @expo/metro-config@0.2.6 does that work?

**Comment by cvburgess** [comment]
Im not sure on the metro config - we were on 1.x with 42.

I upgraded to the latest and everything broke, so I'm isolating.

Do changes to the metro-config package require a new dev client or no?

**Comment by brentvatne** [comment]
https://github.com/expo/expo-cli/blob/master/CHANGELOG.md

**Comment by cvburgess** [comment]
Ok so that combination of things appeared to fix my issues.
Will try bumping metro again shortly

**Comment by brentvatne** [comment]
fixed in expo-cli@4.12.12 and @expo/metro-config@0.2.8

if this doesn't resolve the issue for you then please create a new issue with reproducible example. more info in https://github.com/expo/expo/issues/15035
